tools/FileTestDefs.sh

```diff
@@ -215,6 +215,23 @@ carriage_returns_test() {
 }
 standard_checks+=(carriage_returns)
 
+# Check for carriage returns
+non_ascii_characters() {
+    whitelist "$1" '.png' 'FileTestDefs.sh' 'octicons.css' &&
+    staged_grep -q -P "[^\x00-\x7F]" "$1"
```

Comment:
This is a useless review comment. Please provide an actual change request.

Reply:
@wthrowe would it be preferable to generate a differing stream with something like `sed` or `tr` and compare the streams, or call the Perl one-liner directly, rather than using Perl grep?

Reply:
Perl is certainly portable.  I don't know how the others handle non-ASCII, but if they work everywhere we care about that's fine.

I don't have a mac (the only sort of non-GNU portability I think we care about) to test with.  I just learned a bunch of this stuff while writing shell scripts for SpEC.  You'll need to grab someone who actually has a machine to test on to figure out what works.